src/V3Premit.cpp

```diff
@@ -37,52 +37,6 @@
 
 constexpr int STATIC_CONST_MIN_WIDTH = 256;  // Minimum size to extract to static constant
 
-//######################################################################
-// Structure for global state
-
-class PremitAssignVisitor final : public VNVisitor {
-private:
-    // NODE STATE
-    //  AstVar::user3()         // bool; occurs on LHS of current assignment
-    const VNUser3InUse m_inuser3;
-
-    // STATE
-    bool m_noopt = false;  // Disable optimization of variables in this block
-
-    // METHODS
-    VL_DEBUG_FUNC;  // Declare debug()
-
-    // VISITORS
-    virtual void visit(AstNodeAssign* nodep) override {
-        // AstNode::user3ClearTree();  // Implied by VNUser3InUse
-        // LHS first as fewer varrefs
-        iterateAndNextNull(nodep->lhsp());
-        // Now find vars marked as lhs
-        iterateAndNextNull(nodep->rhsp());
```

Comment:
I think you changed behavior here.  The old code did LHS then RHS, the new code will do RHS than LHS.

Reply:
If you look at the replacement at the use site, this visitor is always invoked on `AstNodeAssign`, and I replaced it with 2 separate `foreach`, first on the LHS, then or the RHS. Strictly, the behaviour is changed if there is a write ref on the RHS that is subsequently also read, but that was kind of bust in the original too, so I think this is ok.